enterprise/backend/test/metabase_enterprise/data_editing/api_test.clj

```diff
@@ -703,3 +703,106 @@
                                                   :params {:status "approved"}}
                                    :query-params {:dashcard-id (:id dashcard)}})
                              (select-keys [:status :body])))))))))))))
+
+(deftest list-and-add-to-dashcard-test
+  (let [list-req #(mt/user-http-request-full-response
+                   (:user % :crowberto)
+                   :get
+                   "ee/data-editing/tmp-action")]
+    (mt/with-premium-features #{:table-data-editing}
+      (mt/test-drivers #{:h2 :postgres}
+        (data-editing.tu/toggle-data-editing-enabled! true)
+        (with-open [test-table (data-editing.tu/open-test-table! {:id 'auto-inc-type
+                                                                  :text      [:text]
+                                                                  :int       [:int]
+                                                                  :timestamp [:timestamp]
+                                                                  :date      [:date]}
+                                                                 {:primary-key [:id]})]
+          (let [{:keys [status body]}
+                (list-req {})
+                all-actions  (:actions body)
+                crud-actions (filter #(= @test-table (:table_id %)) all-actions)
+                {create-action "table.row/create"
+                 update-action "table.row/update"
+                 delete-action "table.row/delete"}
+                (u/index-by :kind crud-actions)]
+            (is (= 200 status))
+            (testing "table actions have neg ids"
+              (is (every? neg? (map :id crud-actions))))
+            (testing "one action for each crud op"
+              (is (= {"table.row/create" 1
+                      "table.row/update" 1
+                      "table.row/delete" 1}
+                     (frequencies (map :kind crud-actions)))))
+            (mt/with-temp [:model/Dashboard dash {}]
+              (let [dashboard-url (str "dashboard/" (:id dash))
+                    card-counter (atom 0)
+                    card-input (fn [action]
```

Comment:
nit: I'd prefer you constructed these with `map-indexed` instead of using an item.

Reply:
57e21d1dee89